src/coupon/README.md

```diff
@@ -0,0 +1,170 @@
+# Coupon System for MCP SDK
+
+This module provides a system for creating, signing, verifying, and managing coupons in the MCP SDK.
+
+## Overview
+
+Coupons are verifiable tokens that prove the legitimacy of requests between clients and servers. They help establish trust and reputation in a decentralized system.
+
+When a client makes a request to a server, the server issues a verifiable `Coupon`, which proves:
+- The request occurred
+- The issuer's identity
+- The recipient's identity
+- The legitimacy of the interaction
+
+## Usage Examples
+
+### Server-side (Creating coupons)
+
+```typescript
+import { Server } from '@modelcontextprotocol/sdk/server';
+import { couponStorage } from '@modelcontextprotocol/sdk/coupon';
+
+// Create a server with coupons enabled
+const server = new Server({
+  name: 'MyServer',
+  version: '1.0.0',
+  enableCoupons: true
+});
+
+// Configure the server's identity for coupon issuance
+server.configureCouponIdentity(
+  {
+    commonName: 'Example Server',
+    organization: 'Example Org',
+    country: 'US'
+  },
+  myServerCertificate,
+  myServerPrivateKey
+);
+
+// Set up a coupon callback
+server.oncoupon = (coupon) => {
```

Comment:
Minor thing but everyone in js community seems to be running away from mutable objects. Can you instead set it like `server.setCouponCallback`?